src/repl/scala/tools/nsc/interpreter/ReplReporter.scala

```diff
@@ -3,23 +3,67 @@
  * @author Paul Phillips
  */
 
-package scala.tools.nsc
-package interpreter
+package scala.tools.nsc.interpreter
 
-import reporters._
-import IMain._
-import scala.reflect.internal.util.{NoPosition, NoSourceFile, Position}
+import java.io.PrintWriter
+
+import scala.tools.nsc.reporters._
+import scala.reflect.internal.util.{NoSourceFile, Position}
+import scala.tools.nsc.Settings
+
+trait ReplReporterConfig {
+  def settings: Settings
+
+  def out: PrintWriter
+
+  // whether to print anything
+  var totalSilence: Boolean
+
+  // whether to print result lines
+  var printResults: Boolean
+
+  def beQuietDuring[T](body: => T): T = {
+    val saved = printResults
+    printResults = false
+    try body
+    finally printResults = saved
+  }
+
+  def beSilentDuring[T](operation: => T): T = {
+    val saved = totalSilence
+    totalSilence = true
+    try operation
+    finally totalSilence = saved
+  }
+
+
+  var maxPrintString: Int
+  def isTruncating: Boolean
+
+  var unwrapStrings: Boolean
+  def unmangle(s: String): String
+
+  // compiler is initialized
+  def initializeComplete: Boolean
+
+  def lastInputSingleLine: Boolean
+  def preambleLineDelta: Int
+}
+
+// TODO: move this out of imain and into iloop
+/** Like ReplGlobal, a layer for ensuring extra functionality. */
+class ReplReporter(config: ReplReporterConfig, colorOk: Boolean = false, indentDepth: Int = 0, trace: Boolean = false)
+  extends ConsoleReporter(config.settings, Console.in, new ReplStrippingWriter(config)) {
+
+  import config._
 
-/** Like ReplGlobal, a layer for ensuring extra functionality.
- */
-class  ReplReporter(intp: IMain) extends ConsoleReporter(intp.settings, Console.in, new ReplStrippingWriter(intp)) {
   def printUntruncatedMessage(msg: String) = withoutTruncating(printMessage(msg))
 
   /** Whether very long lines can be truncated.  This exists so important
    *  debugging information (like printing the classpath) is not rendered
    *  invisible due to the max message length.
    */
-  private var _truncationOK: Boolean = !intp.settings.verbose
+  private var _truncationOK: Boolean = !settings.verbose
```

Comment:
The leading underscore means it could have been initialized with an underscore. Maybe I should add a StackOverflow Q&A for that?